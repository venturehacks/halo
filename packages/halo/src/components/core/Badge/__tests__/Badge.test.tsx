import { render } from 'enzyme';
import React from 'react';

import { Badge } from '../index';

describe('Badge', () => {
  test('smoke', () => {
    const component = render(<Badge>NICE BADGE</Badge>);
    expect(component.text()).toEqual('NICE BADGE');
  });

  describe('snapshot', () => {
    describe('color', () => {
      test('gray', () => {
        const component = render(<Badge />);
        expect(component).toMatchSnapshot();
      });

      test('blue', () => {
        const component = render(<Badge color="blue" />);
        expect(component).toMatchSnapshot();
      });

      test('green', () => {
        const component = render(<Badge color="green" />);
        expect(component).toMatchSnapshot();
      });
    });

    describe('shape', () => {
      test('circle', () => {
        const component = render(<Badge />);
        expect(component).toMatchSnapshot();
      });

      test('square', () => {
        const component = render(<Badge shape="square" />);
        expect(component).toMatchSnapshot();
      });
    });
  });
});
