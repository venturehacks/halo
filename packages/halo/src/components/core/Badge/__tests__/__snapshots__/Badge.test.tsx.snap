// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge snapshot color blue 1`] = `
<div
  class="absolute right-0 border-solid border-slate-100 text-center text-2xs font-medium uppercase leading-none antialiased bg-blue-600 border-blue-100 text-white rounded-full min-w-4 p-1 -top-1"
/>
`;

exports[`Badge snapshot color gray 1`] = `
<div
  class="absolute right-0 border-solid border-slate-100 text-center text-2xs font-medium uppercase leading-none antialiased bg-slate-600 border-slate-100 text-white rounded-full min-w-4 p-1 -top-1"
/>
`;

exports[`Badge snapshot color green 1`] = `
<div
  class="absolute right-0 border-solid border-slate-100 text-center text-2xs font-medium uppercase leading-none antialiased bg-green-600 border-green-100 text-white rounded-full min-w-4 p-1 -top-1"
/>
`;

exports[`Badge snapshot position bottom 1`] = `
<div
  class="absolute right-0 border-solid border-slate-100 text-center text-2xs font-medium uppercase leading-none antialiased bg-slate-600 border-slate-100 text-white rounded-full min-w-4 p-1 bottom-1"
/>
`;

exports[`Badge snapshot position top 1`] = `
<div
  class="absolute right-0 border-solid border-slate-100 text-center text-2xs font-medium uppercase leading-none antialiased bg-slate-600 border-slate-100 text-white rounded-full min-w-4 p-1 -top-1"
/>
`;

exports[`Badge snapshot shape circle 1`] = `
<div
  class="absolute right-0 border-solid border-slate-100 text-center text-2xs font-medium uppercase leading-none antialiased bg-slate-600 border-slate-100 text-white rounded-full min-w-4 p-1 -top-1"
/>
`;

exports[`Badge snapshot shape square 1`] = `
<div
  class="absolute right-0 border-solid border-slate-100 text-center text-2xs font-medium uppercase leading-none antialiased bg-slate-600 border-slate-100 text-white rounded-md min-w-4 p-1 -top-1"
/>
`;

exports[`Badge snapshot size size 1`] = `
<div
  class="absolute right-0 border-solid border-slate-100 text-center text-2xs font-medium uppercase leading-none antialiased bg-slate-600 border-slate-100 text-white rounded-full min-w-6 p-1 -top-2"
/>
`;

exports[`Badge snapshot size sm 1`] = `
<div
  class="absolute right-0 border-solid border-slate-100 text-center text-2xs font-medium uppercase leading-none antialiased bg-slate-600 border-slate-100 text-white rounded-full min-w-4 p-1 -top-1"
/>
`;
